src/main/java/org/openstreetmap/atlas/geography/MultiPolyLine.java

```diff
@@ -88,12 +90,42 @@ public Rectangle bounds()
         return Rectangle.forLocations(locations);
     }
 
+    @Override
+    public boolean equals(final Object other)
+    {
+        if (!(other instanceof MultiPolyLine))
+        {
+            return false;
+        }
+        final MultiPolyLine otherItem = (MultiPolyLine) other;
+        final Set<PolyLine> polyLineSet = new HashSet<>(this.polyLineList);
+        final Set<PolyLine> otherPolyLineSet = new HashSet<>(otherItem.getPolyLineList());
+        return otherItem.getPolyLineList().size() == this.getPolyLineList().size()
```

Comment:
As ```polyLineSet``` & ```otherPolyLineSet``` are HashSet, I think you can directly return ```polyLineSet.equals(otherPolyLineSet)```. equals method for set checks for the size equality.